lumen/ai/views.py

```diff
@@ -108,6 +113,7 @@ async def _render_component(self):
                 output = pn.Column(download_pane, table)
             else:
                 output = self.component.__panel__()
+            pn.state.cache[self.spec] = output
```

Comment:
This is a giant memory leak.

Reply:
There was some discussion around having a `state.session_cache` but for now you'll likely have to add a WeakKeyDictionary indexed by the `pn.state.curdoc` somewhere, e.g. `lumen.state`

```
_session_cache = WeakKeyDictionary()
_global_cache = {}
...
def session_cache(self):
    doc = pn.state.curdoc
    if doc is None:
        return self._global_cache
    elif doc not in self._session_cache:
        self._session_cache[doc] = {}
    return self._session_cache[doc]      
```

Reply:
Can we use the existing memory implementation?
```


class _Memory(Viewer):

    _session_contexts: ClassVar[WeakKeyDictionary[Document, Any]] = WeakKeyDictionary()

    _views: ClassVar[WeakKeyDictionary[Document, Any]] = WeakKeyDictionary()

    _global_view = None

    _global_context = {}

    @property
    def _curcontext(self):
        if state.curdoc:
            if state.curdoc in self._session_contexts:
                context = self._session_contexts[state.curdoc]
            else:
                self._session_contexts[state.curdoc] = context = {}
            return context
        else:
            return self._global_context
```

Reply:
Hmm, technically yes but I'd still say no, the memory is meant to be used by the LLM, not for our code to cache things.